likecoin/js/admin/likecoin_metabox.js

```diff
@@ -56,35 +56,128 @@ async function onISCNCallback(event) {
-  const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${ipfsHash}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
-  window.open(likeCoISCNWidget, '_blank', 'menubar=no,location=no,width=576,height=768');
-  window.addEventListener('message', onISCNCallback, false);
+  try {
+    if (!mattersIPFSHash && !arweaveIPFSHash) {
+      throw new Error('NO_IPFS_HASH_FOUND');
+    }
+    const titleString = encodeURIComponent(title);
+    const tagsArray = tags || [];
+    const tagsString = tagsArray.join(',');
+    const urlString = encodeURIComponent(url);
+    const redirectString = encodeURIComponent(siteurl);
+    let fingerprint = '';
+    if (mattersIPFSHash) {
+      const mattersIPFSHashFingerprint = `ipfs://${mattersIPFSHash}`;
+      if (fingerprint) {
+        fingerprint = fingerprint.concat(`,${mattersIPFSHashFingerprint}`);
+      } else {
+        fingerprint = mattersIPFSHashFingerprint;
+      }
+    }
+    if (arweaveIPFSHash) {
+      const arweaveIPFSHashFingerprint = `ipfs://${arweaveIPFSHash}`;
+      if (fingerprint) {
+        fingerprint = fingerprint.concat(`,${arweaveIPFSHashFingerprint}`);
+      } else {
+        fingerprint = arweaveIPFSHashFingerprint;
+      }
+    }
+    if (arweaveId) {
+      const arweaveFingerprint = `ar://${arweaveId}`;
+      fingerprint = fingerprint.concat(`,${arweaveFingerprint}`);
+    }
+    const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${fingerprint}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
+    window.open(likeCoISCNWidget, '_blank', 'menubar=no,location=no,width=576,height=768');
+    window.addEventListener('message', onISCNCallback, false);
+  } catch (error) {
+    console.log(`error occured when submitting ISCN: ${error}`);
+  }
 }
 
+async function onEstimateAndUploadArweave(e) {
+  e.preventDefault();
+  try {
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/estimate`,
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    const {
+      ipfsHash, LIKE, memo, arweaveId,
+    } = res;
+    if (arweaveId && ipfsHash) { // skip Arweave upload flow
+      onSubmitToISCN();
+      return;
+    }
+    const { siteurl } = wpApiSettings;
+    const memoString = encodeURIComponent(memo);
+    const redirectString = encodeURIComponent(siteurl);
+    const likePayWidget = `https://like.co/in/widget/pay?to=like-arweave&amount=${LIKE}&remarks=${memoString}&opener=1&redirect_uri=${redirectString}`;
+    window.open(
+      likePayWidget,
+      '_blank',
+      'menubar=no,location=no,width=576,height=768',
+    );
+    window.addEventListener(
+      'message',
+      onLikePayCallbackAndUploadArweave,
+      false,
+    );
```

Comment:
what if user close the window and open it again?

Reply:
Use removeEventListener to avoid memory over-usage.